src/main/java/com/hillel/items_exchange/service/PhoneService.java

```diff
@@ -0,0 +1,37 @@
+package com.hillel.items_exchange.service;
+
+import com.hillel.items_exchange.exception.UnprocessableEntityException;
+import com.hillel.items_exchange.model.Phone;
+import com.hillel.items_exchange.model.User;
+import com.hillel.items_exchange.util.MessageSourceUtil;
+
+import org.springframework.beans.BeanUtils;
+import org.springframework.stereotype.Service;
+
+import lombok.RequiredArgsConstructor;
+
+import java.util.List;
+
+@RequiredArgsConstructor
+@Service
+public class PhoneService {
+    private final MessageSourceUtil messageSourceUtil;
+
+    public void updateAll(List<Phone> updatedPhones, User currentUser) {
+        updatedPhones.forEach(phone -> {
+            if (phone.getId() == 0) {
+                phone.setUser(currentUser);
+                currentUser.getPhones().add(phone);
+            } else {
+                Phone existedPhone = currentUser.getPhones().stream()
+                        .filter(p -> p.getId() == phone.getId())
+                        .findAny()
+                        .orElseThrow(() -> new UnprocessableEntityException(
+                                messageSourceUtil.getExceptionMessageSource("exception.invalid.dto")));
+                BeanUtils.copyProperties(phone, existedPhone, "id", "user");
+            }
+        });
+        currentUser.getPhones().removeIf(phone ->
+                updatedPhones.stream().map(Phone::getId).noneMatch(id -> id == phone.getId()));
+    }
```

Comment:
If you add PhoneRepository, you can do something like this (just my opinion):

public void updateAll(List<Phone> newPhoneList, User user) {
    newPhoneList.forEach(phone -> {
        if (phone.getId() == 0) {
            phoneRepository.saveAndFlush(phone);
            user.addPhone(phone);
        } else {
            Phone existedPhone = phoneRepository.findById(phone.getId()).orElseThrow(() ->
                    new UnprocessableEntityException(messageSourceUtil.getExceptionMessageSource("exception.invalid.dto")));
            BeanUtils.copyProperties(phone, existedPhone, "id", "user");
            phoneRepository.saveAndFlush(existedPhone);
        }
    });
    user.getPhones().removeIf(phone -> !newPhoneList.contains(phone));

Reply:
Yes, your example works, but imagine we got there UnprocessableEntityException here and operation of updating is interrupted. The previous code made changes (if they were) in User entity, Child table, so we have no transaction while updating user info.
I tried to save all entities saving User entity, but not erlier.